stacks/upload-db-stack.js

```diff
@@ -105,8 +112,14 @@ export function UploadDbStack({ stack, app }) {
    */
   const spaceMetricsTable = new Table(stack, 'space-metrics', spaceMetricsTableProps)
 
+  /**
+   * This table tracks storage providers in the system.
+   */
+  const storageProviderTable = new Table(stack, 'storage-provider', storageProviderTableProps)
```

Comment:
Every storage provider has a DID, a URL and a proof (a delegation to the upload service allowing `blob/allocate` and `blob/accept` invocations). They also have a weight, which allows some providers to be picked more often than others.